### ISSUE TITLE
Broken/missing handler on Directory Channel view

### ISSUE THREAD

**Issue by stephen-cox** [localgovdrupal/localgov_directories#228]
On fresh installs, after enabling directories, the contextual filters are broken; it just says Broken/missing handler. This means that once the Demo module is enabled all directory items are listed on all directory channels.

The message from the broken handler is 

```
Configure contextual filter: Broken/missing handler

The handler for this item is broken or missing. The following details are available:

    id: localgov_directory_channels
    table: search_api_index_localgov_directories_index_default
    field: localgov_directory_channels
    relationship: none
    group_type: group
    admin_label:
    plugin_id: search_api
    default_action: not found
    title_enable:
    title:
    default_argument_type: fixed
    default_argument_skip_url:
    specify_validation:
    break_phrase:
    not: 

Enabling the appropriate module may solve this issue. Otherwise, check to see if there is a module update available.
```

**Comment by stephen-cox** [comment]
It appears the Directory channels, Content datasource is missing from the available contextual filters.

**Comment by stephen-cox** [comment]
@ekes It looks like this problem doesn't effect the 2.2.0 release. Has something changed in spinning out the localgov_directories_db module that would remove the Content datasource: Directory channels, contextual filter?

**Comment by finnlewis** [comment]
I also see this on a fresh install. 

1. Enable  localgov_directories_promo_page
2. Create a directory facet
3. Create a new directory channel
4. Select facet type 
5. Select promo page as a content type

```
The website encountered an unexpected error. Please try again later.

Drupal\Component\Plugin\Exception\PluginNotFoundException: The "search_api:views_embed__localgov_directory_channel__node_embed" plugin does not exist. Valid plugin IDs for Drupal\facets\FacetSource\FacetSourcePluginManager are: in Drupal\Core\Plugin\DefaultPluginManager->doGetDefinition() (line 53 of core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php).

Drupal\Core\Plugin\DefaultPluginManager->getDefinition('search_api:views_embed__localgov_directory_channel__node_embed') (Line: 16)

```

**Comment by ekes** [comment]
I _now_ have failing tests for the sort field too. I'll add some more as well to make sure we always have these fields.

Seems the fields aren't being imported when the search is disabled, really I don't see why they should be imported when it's enabled but they aren't associated with a content type, so we've probably been lucky so far.

I'm separating these out into more fields, like the existing facet one, that get added when field configuration is inserted. Refactoring the already a bit messy (built up over time) code for field_config_insert and delete.

**Comment by ekes** [comment]
https://github.com/localgovdrupal/localgov_directories/pull/233

Amusingly that test that wasn't failing when we merged the changes is still failing now it works in manual testing :thinking:

**Comment by stephen-cox** [comment]
Fixed with #233
